src/app/portlet/terra-portlet.component.ts

```diff
@@ -1,18 +1,89 @@
 import {
     Component,
-    Input
+    EventEmitter,
+    Input,
+    OnChanges,
+    Output,
+    SimpleChanges
 } from '@angular/core';
+import {
+    animate,
+    state,
+    style,
+    transition,
+    trigger
+} from "@angular/animations";
 
 @Component({
-               selector: 'terra-portlet',
-               styles:   [require('./terra-portlet.component.scss')],
-               template: require('./terra-portlet.component.html')
-           })
-export class TerraPortletComponent
+    selector: 'terra-portlet',
+    styles:   [require('./terra-portlet.component.scss')],
+    template: require('./terra-portlet.component.html'),
+    animations: [
+        trigger('collapsedState', [
+            state('collapsed', style({
+                height: 0,
+                'padding-top': 0,
+                'padding-bottom': 0
+            })),
+            state('expanded', style({
+                height: '*'
+            })),
+            transition('collapsed <=> expanded', [
+                animate(100)
+            ])
+        ])
+    ]
+})
+export class TerraPortletComponent implements OnChanges
 {
+
     @Input() inputPortletHeader:string;
 
+    @Input() isCollapsable: boolean = false;
+
+    @Input() collapsed: boolean = false;
+
+    @Output() collapsedChange: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
Die Benennung von `collapsed` und `collapsedChange` ermöglicht bei der Verwendung der Komponente ein Two-Way-Binding auf `[(collapsed)]=myCollapsed`.
https://angular-2-training-book.rangle.io/handout/components/app_structure/two_way_data_binding.html

Eine Umbenennung in `inputCollapsed` und `outputCollapsedChange` würde diese Möglichkeit verbauen. Vielleicht ist hier nochmal eine Diskussion der Code-Richtlinien erforderlich.